src/librustc_data_structures/graph_algorithms/dominators/mod.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use super::Graph;
+use super::iterate::reverse_post_order;
+use super::node_vec::NodeVec;
+
+use std::fmt;
+
+#[cfg(test)]
+mod test;
+
+pub fn dominators<G: Graph>(graph: &G)
```

Comment:
we ought to cite this paper, which is the origin of the algorithm (my bad):

https://www.cs.rice.edu/~keith/EMBED/dom.pdf